### PR TITLE
Avoid double-scoring of CV models in AutoML 

### PR DESCRIPTION
**Pull request by michalkurka**

### CODE REVIEW

File: h2o-algos/src/main/java/hex/ensemble/StackedEnsembleModel.java [hunk 1]
```diff
@@ -183,23 +183,43 @@ protected void map(int id) {
...
+    }
+
+    @Override
+    public ModelMetrics getOrMakeMetrics(Frame fr, Frame adaptFrm) {
+      return _modelMetrics;
+    }
+
+    @Override
+    public ModelMetrics.MetricBuilder<?> getOrMakeMetricBuilder(Frame adaptFrm) {
+      return StackedEnsembleModel.this.scoreMetrics(adaptFrm);
```
Comment (tomasfryda) [inline h2o-algos/src/main/java/hex/ensemble/StackedEnsembleModel.java:219]: I don't think you have to override this method. 

We don't have working implementation `StackedEnsembleModel#makeMetricBuilder` but also we don't have working implementation of `StackedEnsembleModel#score0` which is used by `BigScore#map` and using that by `Model#scoreMetrics`. 

Either way I think this call would not work so I'd rather keep it consistent with the parent class (and eventually make SE more like the rest of the models).

Comment (michalkurka) [reply h2o-algos/src/main/java/hex/ensemble/StackedEnsembleModel.java:219]: @tomasfryda thanks for the comment, I get it now, there is no CV for Stacked Ensembles as such, there is CV on the meta learner level but not on the outer SE model

this means I can simplify the code quite a bit!

File: h2o-core/src/main/java/hex/Model.java [hunk 2]
```diff
@@ -1859,12 +1861,44 @@ protected Frame predictScoreImpl(Frame fr, Frame adaptFrm, String destination_ke
...
+      _mb = mb;
+      _rawPreds = rawPreds;
+      _outputPreds = outputPreds;
+    }
+    
+    public final Frame getPredictions() {
+      return _outputPreds;
+    }
+
+    public ModelMetrics.MetricBuilder<?> getMetricBuilder(Frame adaptFrm) {
```
Comment (tomasfryda) [inline h2o-core/src/main/java/hex/Model.java:1890]: `Frame adaptFrm` is unused. Is it a remnant of the previous version (IIRC it was used in `StackedEnsemblePredictScoreResult`)?

Comment (michalkurka) [reply h2o-core/src/main/java/hex/Model.java:1890]: good catch, I need to remove this argument as well

**Review by tomasfryda — APPROVED** [review]
Looks good to me. Thank you @michalkurka!

**Comment by michalkurka** [comment]
@tomasfryda thanks for the review and catching the unnecessary complexity built for SE

**Pull request merged**